main.cpp

```diff
@@ -66,54 +68,22 @@ Triangles triangles_from_scene(const aiScene* scene) {
     return triangles;
 }
 
-static const char USAGE[] =
-    R"(Usage: raytracer <filename> [options]
-
-Options:
-  -w --width=<px>                   Width of the image [default: 640].
-  -a --aspect=<num>                 Aspect ratio of the image. If the model has
-                                    specified the aspect ratio, it will be
-                                    used. Otherwise default value is 1.
-  -d --max-depth=<int>              Maximum recursion depth for raytracing
-                                    [default: 3].
-  --shadow=<float>                  Intensity of shadow [default: 0.5].
-  --background=<3x float>           Background color [default: 0 0 0].
-  -p --pixel-samples=<int>          Number of samples per pixel [default: 1].
-  -m --monte-carlo-samples=<int>    Monto Carlo samples per ray [default: 8].
-                                    Used only in pathtracer.
-  -t --threads=<int>                Number of threads [default: 1].
-  --inverse-gamma=<float>           Inverse of gamma for gamma correction
-                                    [default: 0.454545].
-  --no-gamma-correction             Disables gamma correction.
-  --max-visibility=<float>          Any object farther away is dark. [default: 2.0]
-                                    Used only in raycaster.
-  --exposure=<float>                Exposure [default: 1].
-)";
+// Defined in the file with the trace implementation for the corresponding
+// renderer.
+extern const char* USAGE;
```

Comment:
Not really, we still would need a good parser, which would fail if the user inputs wrong parameters. I think it is nice to have the configuration, where it should be. Note that it is also different for different renderer, even if the config class is the same.